### PR TITLE
LPS-98197 DLFileEntryTrashHandlerTest and DLFileEntryFinderTest

### PR DESCRIPTION
**Pull request by Preston-Crary**
There's some other failing integration tests with StaleObjectStateExceptions:

1. WorkflowTaskManagerImplTest
2. DLFolderUADAnonymizerTest
3. RepositoryModelUtilTest
4. FolderStagedModelDataHandlerTest
5. FileEntryStagedModelDataHandlerTest

Maybe some others as well. I'll run the tests here so we can see if any of those were resolved by these changes.  Can you use these three fixes as examples of how to handle the other ones?

If you get stuck let me know and I'll try to help when I can.

### CODE REVIEW

**Comment by Preston-Crary** [comment]
ci:test:backend-hypersonic

**Comment by vicnate5** [comment]
CC @achaparro

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Comment by AliciaGarciaGarcia** [comment]
Pull request submitted to https://github.com/Preston-Crary/liferay-portal/pull/804. [See changes here.](https://github.com/AliciaGarciaGarcia/liferay-portal/compare/e79eada3d488~1...e79eada3d488)

**Comment by AliciaGarciaGarcia** [comment]
Thanks for the possible solution @Preston-Crary, it seems to solve a lot of the test failures and can be a patch. 

The problem is that we want to solve the root cause, and know why now, with the mvccVersion, we are having issues with the hibernate dirty merge and why the DLFileEntry is not the last entry every time.

As you pointed out, in `DLFileEntryLocalServiceImpl`, on the _validateFile_ method the `dlFileEntryPersistence.fetchByG_F_T` is causing the flush of the DLFileEntry, and we can't find why this is happening, and we need the validation of the file because we are breaking the expected functionality (I made 2 new test to check that)


//cc @adolfopa

please see https://github.com/Preston-Crary/liferay-portal/pull/804

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -1962,9 +1959,6 @@ public DLFileEntry updateStatus(
 
 		// File entry
 
-		DLFileEntry dlFileEntry = dlFileEntryPersistence.findByPrimaryKey(
-			dlFileVersion.getFileEntryId());
```
Comment (4lejandrito) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java]: Hi @Preston-Crary,

I agree with other changes you've made in the form of `fileEntry = update(fileEntry...)`. It is more correct to use the new object than reuse the potentially stale one. However I don't fully understand why this method needs to change. We were not reusing any objects (it received just the ids) and the only operation that was being performed between the find (`dlFileEntryPersistence.findByPrimaryKey(`) and the update (`dlFileEntryPersistence.update(dlFileEntry);`) was the versions find. I would have thought that `dlFileVersionPersistence.findByF_S(`was to blame here and the solution should be applied inside it.

Please let us know your thoughts.

Thanks for your time!

**Pull request closed without merging**